lib/deploy_pin/collector.rb

```diff
@@ -56,5 +40,25 @@ def exacutable
         task if _tasks[0..index].none? { |_task| task.eql?(_task) }
       end.compact
     end
+
+    def tasks_count
+      tasks.count
+    end
+
+    private
+
+      def files
+        Dir["#{DeployPin.tasks_path}/*.rb"]
+      end
+
+      def tasks
+        files.map do |file|
+          task = DeployPin::Task.new(file)
+          task.parse_file
+
+          # check if task passed criteria
+          task if DeployPin::TaskCriteria.new(identifiers: identifiers, task: task).passed?
```

Comment:
no need to instantiate criteria each time (identifiers are global for all tasks) leave just identifiers in initializer, move it before the loop & pass :task parameter to passed?(:task)